chain/types/ethtypes/errors.go

```diff
@@ -0,0 +1,43 @@
+package ethtypes
+
+import (
+	"fmt"
+
+	"golang.org/x/xerrors"
+)
+
+var ErrExecutionReverted = xerrors.New("execution reverted")
```

Comment:
You've made 2 error objects in this file (this one, and the `struct` that conforms to the interface) and the only time you use this one is for its string in this file and once in the itest. I'd just either use this string, `"execution reverted"` inline in the constructor, or make it a private `const` here. It's OK to expect `ErrorContains` `"execution reverted"` in the test without linking it to an existing const, in fact that's desirable in a test so make sure what you think is a const is actually what you want it to be.